source.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @module  web-audio-stream/source
+ *
+ * Source pull-stream for web-audio node
+ */
+'use strict';
+
+
+var pull = require('pull-stream/pull');
+var asyncMap = require('pull-stream/throughs/async-map');
+var infinite = require('pull-stream/sources/infinite');
+var createReader = require('./reader');
+
+module.exports = source;
+
+
+function source (node, options) {
+	let read = createReader(node, options);
+
+	//FIXME: too clumsy. I’d expect pull.asyncSource(read)
```

Comment:
nice! i agree there should be something more graceful, _pull-async_ was meant for a specific use case (turning a single continuable into a stream, kinda like an async version of `pull.once`), keen for a re-usable module that does what you're doing.